core/string/src/str.rs

```diff
@@ -291,6 +291,16 @@ impl PartialEq for JsStr<'_> {
     }
 }
 
+impl PartialEq<&str> for JsStr<'_> {
+    #[inline]
+    fn eq(&self, other: &&str) -> bool {
+        match self.variant() {
+            JsStrVariant::Latin1(v) => v == other.as_bytes(),
+            JsStrVariant::Utf16(v) => other.encode_utf16().zip(v).all(|(a, b)| a == *b),
+        }
+    }
+}
+
```

Comment:
Can you check if `PartialEq<str>` is also implemented, and add it if it's not? Most std collections implement for both `[T]` and `&[T]` just in case someone uses `PartialEq::eq` directly.

Reply:
Done